File: csharp/ToolGood.Algorithm/Internals/Functions/Value/Function_Number.cs
using System;
using System.Globalization;
using System.Text;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.Value
{
	internal sealed class Function_Number : Function_0
	{
		private readonly Operand _value;

		public Function_Number(Operand value)
		{
			_value = value;
		}

		public override string Name => "Num";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter = null)
		{
			return _value;
		}

		public override OperandType GetResultType()
		{
			return OperandType.NUMBER;
		}

		public override void ToString(StringBuilder stringBuilder, bool addBrackets)
		{
			// 必须使用 InvariantCulture, 否则在 de-DE 等区域下小数点会输出为逗号, 导致 ToString 结果无法被本引擎重新解析
			stringBuilder.Append(_value.NumberValue.ToString(CultureInfo.InvariantCulture));
		}
	}
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/internals/functions/value/Function_Number.java
package toolgood.algorithm.internals.functions.value;

import java.util.function.BiFunction;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_Number extends Function_0 {
    private final Operand _value;

    public Function_Number(Operand value) {
        _value = value;
    }

    @Override
    public String Name() {
        return "Num";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        return _value;
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.NUMBER;
    }

    @Override
    public void ToString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append(_value.NumberValue());
    }
}
